sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -423,6 +423,47 @@ AZ_NODISCARD az_result az_iot_provisioning_client_query_status_get_publish_topic
     size_t mqtt_topic_size,
     size_t* out_mqtt_topic_length);
 
+/**
+ * @brief Azure IoT Provisioning Client options for az_iot_provisioning_client_get_request_payload.
+ *        Not currently used.  Reserved for future use.
+ *
+ */
+struct az_iot_provisioning_client_payload_options;
+
```

Comment:
how about defining typedef that points to void for now and then in latter we point that typedef to some memory type?


typedef void az_iot_provisioning_client_payload_options;

Reply:
I like this - cleaner and it removes all the `struct` keywoard all over place too.  Fixing.